### DISCUSSION TITLE
Need help to configure in home assistant

### DISCUSSION THREAD

**Discussion by Malli88** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#439] [Q&A] [answered]
Hi Community,

I'm pretty new in HA and I need a little bit of help.
What I have:
- HA installation in a VM in my Unraid
- Roborock S7 with Dust Collector Dock

What I have done so fare:
- Installation of lovelace frontend with HA Community Apps
- Download of the latest xiaomi-vacuum-map-card.js to /config/www/custom_lovelace/xiaomi-vacuum-map-card.js
- created a ressource 
![image](https://user-images.githubusercontent.com/7073723/179833307-e6013ae8-8dca-4f63-942d-3fcd747e8457.png)

Now I need some help to get this thing up and running. I have no idea to bring that on my dashboard.
Would be very nice if someone could help me.

Thank you!

Cheers
Stephan

**Comment by PiotrMachowski** [comment, marked as answer]
You have to use just one installation method, I highly suggest using HACS. Once you have it installed you should be able to add this card using UI editor

**Reply by Malli88** [reply]
Hi @PiotrMachowski ,

I used the instructions you mentioned in your manual.
Unfortunately, I couldn't find any Card in the dashboard edit.
Any Idea?

**Reply by PiotrMachowski** [reply]
Do you have [HACS](https://hacs.xyz/) installed? 

[Here](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card#hacs) I have shortly described how to install this card using HACS (which is a recommended way to install it instead of manually)

**Reply by Malli88** [reply]
You mean that?
![image](https://user-images.githubusercontent.com/7073723/179840213-89ea56a1-e774-4fec-800c-df7c33b171c3.png)

**Reply by PiotrMachowski** [reply]
Yes, it looks ok. You can remove `config/www/custom_lovelace/xiaomi-vacuum-map-card.js` file and its entry in resources (there should be this one left: `hacsfiles/lovelace-xiaomi-vacuum-map-card/xiaomi-vacuum-map-card.js`). Restart HA and you should be able to find this card in UI editor (you can open it in upper right corner of dashboard)

**Reply by Malli88** [reply]
ok, where I can find the configuration.yaml to add my camera configuration?
I have the token and I installed the Xiaomi Cloud Map Extractor

**Reply by PiotrMachowski** [reply]
You have to install an addon to do it, I recommend "Studio Code Server"

**Reply by Malli88** [reply]
![image](https://user-images.githubusercontent.com/7073723/179846406-06d6c42a-13d4-437b-afd5-0111da1840bb.png)
Here?

**Reply by Malli88** [reply]
something like that
![image](https://user-images.githubusercontent.com/7073723/179846950-b1ae7c6f-9075-42aa-bb47-ba60f6ba237c.png)

**Reply by PiotrMachowski** [reply]
The placing is correct, but you have to use `secrets.yaml` to store host/token/username/password: https://www.home-assistant.io/docs/configuration/secrets/

**Reply by Malli88** [reply]
WOOP WOOP
![image](https://user-images.githubusercontent.com/7073723/179848647-b9a8a5de-a169-4edf-8c6b-d729fe7cfe14.png)

**Reply by Malli88** [reply]
Thank you so much!

**Reply by PiotrMachowski** [reply]
You can also check out [this guide](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/discussions/317) to set up room cleaning
